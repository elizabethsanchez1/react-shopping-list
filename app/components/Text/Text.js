import React from 'react';
import PropTypes from 'prop-types';
import { Text as BaseText, StyleSheet } from 'react-native';
import theme from "../../../app/styles/theme.style";


const styles = StyleSheet.create({
  header: {
    fontSize: theme.FONT_SIZE_HEADERBAR,
    fontFamily: theme.PRIMARY_FONT_FAMILY,
    fontWeight: theme.FONT_WEIGHT_MEDIUM,
    color: theme.PRIMARY_FONT_COLOR,
  },
  standardText: {
    fontFamily: theme.PRIMARY_FONT_FAMILY,
    fontWeight: theme.FONT_WEIGHT_MEDIUM,
    fontSize: theme.FONT_SIZE_MEDIUM,
    color: theme.PRIMARY_FONT_COLOR,
  },
  small: {
    fontSize: theme.FONT_SIZE_SMALL,
  },
  medium: {
    fontSize: theme.FONT_SIZE_MEDIUM,
  },
  large: {
    fontSize: theme.FONT_SIZE_LARGE,
  },
  lightFontWeight: {
    fontWeight: theme.FONT_WEIGHT_LIGHT,
  },
  mediumFontWeight: {
    fontWeight: theme.FONT_WEIGHT_MEDIUM,
  },
  boldFontWeight: {
    fontWeight: theme.FONT_WEIGHT_BOLD,
  }
});


// TODO: Font weight does not work for some reason???

const Text = ({ children, size, weight, textStyling }) => {
  let styleObject = { ...styles.standardText  };

  if ( size === 'small' ) {
    styleObject = { ...styleObject, ...styles.small };
  }

  if ( size === 'medium' ) {
    styleObject = { ...styleObject, ...styles.medium };
  }

  if ( size === 'large' ) {
    styleObject = { ...styleObject, ...styles.large };
  }

  if ( size === 'header' ) {
    styleObject = { ...styleObject, ...styles.header };
  }

  // if ( weight === 'light' ) {
  //   styleObject = { ...styleObject, ...styles.lightFontWeight }
  // }
  //
  // if ( weight === 'medium' ) {
  //   styleObject = { ...styleObject, ...styles.mediumFontWeight }
  // }
  //
  // if ( weight === 'bold' ) {
  //   styleObject = { ...styleObject, ...styles.boldFontWeight }
  // }

  styleObject = { ...styleObject, ...textStyling };

  return (
    <BaseText
      style={ styleObject }
    >
      { children }
    </BaseText>
  )
};

Text.propTypes = {
  children: PropTypes.any,
  size: PropTypes.string,
  textStyling: PropTypes.object,
};

export default Text;
