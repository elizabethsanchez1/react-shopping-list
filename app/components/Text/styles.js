import { create } from 'react-native-platform-stylesheet';
import theme from '../../styles/theme.style';

export default create({
  header: {
    fontSize: theme.FONT_SIZE_HEADERBAR,
    fontFamily: theme.PRIMARY_FONT_FAMILY,
    fontWeight: theme.FONT_WEIGHT_MEDIUM,
    color: theme.PRIMARY_FONT_COLOR,
  },

  // Text specific styles

  standardText: {
    fontFamily: theme.PRIMARY_FONT_FAMILY,
    fontWeight: theme.FONT_WEIGHT_MEDIUM,
    fontSize: theme.FONT_SIZE_MEDIUM,
    color: theme.PRIMARY_FONT_COLOR,
  },
});
