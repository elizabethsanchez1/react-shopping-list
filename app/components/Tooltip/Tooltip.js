import React, { Component } from 'react';
import { Text, StyleSheet, TouchableOpacity, View } from 'react-native';
import PropTypes from 'prop-types';
import BaseToolTip from 'react-native-walkthrough-tooltip';
import { Badge } from 'react-native-elements';
import theme from '../../styles/theme.style';


const styles = StyleSheet.create({
  container: {
    width: 25,
    height: 25,
    borderRadius: 4,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 3 },
    shadowOpacity: 0.8,
    shadowRadius: 2,
    elevation: 1,
  },
  badge: {
    borderRadius: 50,
    width: 25,
    height: 25,
    backgroundColor: theme.PRIMARY_COLOR,
    padding: 0
  }
});

const Tooltip = ( { visibleState, close, open, text, containerStyling, placement } ) => (
  <View style={ [ styles.container, containerStyling ] }>
    <BaseToolTip
      animated
      backgroundColor='transparent'
      isVisible={ visibleState }
      content={ <Text>{ text }</Text> }
      placement={ placement || 'top' }
      onClose={ () => close() }
    >
      <TouchableOpacity onPress={ () => (visibleState) ? close() : open() }>
        <Badge
          value={ '?' }
          containerStyle={ styles.badge }
        />
      </TouchableOpacity>
    </BaseToolTip>
  </View>
);


Tooltip.propTypes = {
  containerStyling: PropTypes.object,
  visibleState: PropTypes.bool,
  close: PropTypes.func,
  open: PropTypes.func,
  text: PropTypes.string,
};


export default Tooltip;
