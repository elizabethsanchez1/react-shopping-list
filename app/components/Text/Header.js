import PropTypes from 'prop-types';
import React from 'react';
import { Text } from 'react-native';
import styles from './styles';

const Header = (props) => (
  <Text
    style={[styles.header, props.style]}
  >
    {props.children}
  </Text>
);

Header.propTypes = {
  children: PropTypes.any,
  style: PropTypes.any,
};

export default Header;
