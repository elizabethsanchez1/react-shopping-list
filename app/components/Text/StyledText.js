import React from 'react';
import PropTypes from 'prop-types';
import { Text } from 'react-native';
import styles from './styles';

const StyledText = (props) => (
  <Text
    style={[styles.standardText, props.style]}
  >
    {props.children}
  </Text>
);


StyledText.propTypes = {
  children: PropTypes.any,
  style: PropTypes.any,
};

export default StyledText;
