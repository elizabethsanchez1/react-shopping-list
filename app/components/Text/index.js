/* eslint-disable import/prefer-default-export */
import Header from './Header';
import StyledText from './StyledText';
import Text from './Text';

export { Header, StyledText, Text };
