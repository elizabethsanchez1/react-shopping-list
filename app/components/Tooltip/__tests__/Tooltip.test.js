import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { Text } from "../../Text";
import Tooltip from '../Tooltip';

describe( 'Tool tip Component', () => {


  it( 'Tooltip visually looks the same', () => {
    const tree = renderer.create(
      <Tooltip
        visibleState={true}
        close={ () => console.log('close') }
        text={<Text textStyling={{ color: 'black' }}>Tooltip content here</Text>}
      />
    ).toJSON();

    expect( tree ).toMatchSnapshot();
  } );

} );




