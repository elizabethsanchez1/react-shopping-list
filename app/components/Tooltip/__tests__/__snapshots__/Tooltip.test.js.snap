// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tool tip Component Tooltip visually looks the same 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 4,
        "elevation": 1,
        "height": 25,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 0.8,
        "shadowRadius": 2,
        "width": 25,
      },
      undefined,
    ]
  }
>
  <View>
    <Modal
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      transparent={true}
      visible={false}
    >
      <View
        accessible={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            0,
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "position": "absolute",
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.8,
              "shadowRadius": 2,
            }
          }
        >
          <View
            style={
              Object {
                "borderBottomColor": "transparent",
                "borderBottomWidth": 9,
                "borderLeftColor": "transparent",
                "borderLeftWidth": 9,
                "borderRightColor": "transparent",
                "borderRightWidth": 9,
                "borderTopColor": "#fff",
                "borderTopWidth": 9,
                "height": 18,
                "left": -9,
                "position": "absolute",
                "top": -9,
                "transform": Array [
                  Object {
                    "rotate": "0deg",
                  },
                  Object {
                    "scale": 0,
                  },
                ],
                "width": 18,
              }
            }
          />
          <View
            onLayout={[Function]}
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 4,
                "padding": 8,
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                  Object {
                    "translateY": 0,
                  },
                  Object {
                    "scale": 0,
                  },
                ],
              }
            }
          >
            <Text>
              <Text
                style={
                  Object {
                    "color": "black",
                    "fontFamily": "RobotoCondensed-Regular",
                    "fontSize": 16,
                    "fontWeight": "600",
                  }
                }
              >
                Tooltip content here
              </Text>
            </Text>
          </View>
        </View>
        <View
          pointerEvents="auto"
          style={
            Object {
              "alignItems": "center",
              "height": 0,
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "top": 0,
              "width": 0,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  undefined,
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#43484d",
                      "borderRadius": 20,
                      "justifyContent": "center",
                      "padding": 12,
                      "paddingBottom": 3,
                      "paddingTop": 3,
                    },
                    Object {
                      "backgroundColor": "#364590",
                      "borderRadius": 50,
                      "height": 25,
                      "padding": 0,
                      "width": 25,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "fontSize": 14,
                      },
                      undefined,
                    ]
                  }
                >
                  ?
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </Modal>
    <View
      onLayout={[Function]}
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#43484d",
                  "borderRadius": 20,
                  "justifyContent": "center",
                  "padding": 12,
                  "paddingBottom": 3,
                  "paddingTop": 3,
                },
                Object {
                  "backgroundColor": "#364590",
                  "borderRadius": 50,
                  "height": 25,
                  "padding": 0,
                  "width": 25,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 14,
                  },
                  undefined,
                ]
              }
            >
              ?
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
